src/main/java/io/vavr/jackson/datatype/deserialize/MaplikeDeserializer.java

```diff
@@ -63,15 +73,8 @@
                                                        TypeDeserializer elementTypeDeserializer,
                                                        JsonDeserializer<?> elementDeserializer);
 
-    @SuppressWarnings("unchecked")
     @Override
     public void resolve(DeserializationContext ctxt) throws JsonMappingException {
-        JavaType keyType = mapType.getKeyType();
-        if (Comparable.class.isAssignableFrom(keyType.getRawClass())) {
-            keyComparator = (Comparator<Object> & Serializable) (o1, o2) -> ((Comparable<Object>) o1).compareTo(o2);
```

Comment:
By the way, @ruslansennov , is there a particular reason to cast the type `Serializable` here? I want to remove it to simplify the logic because it seems to be unnecessary. But I would like to hear from you first.

Reply:
@mincong-h as far as I remember, sortable vavr collections use only serializable comparators to be serializable itself. You can check this by converting json to vavr's HashMap and then applying the java serialization mechanism to it.

Reply:
Thanks for your explanation, @ruslansennov . Let's keep the logic as it is then.